RoslynPluginGenerator/NuGetPackageHandler.cs

```diff
@@ -58,9 +68,9 @@ public FileConflictResolution ResolveFileConflict(string message)
             }
         }
 
-        public NuGetPackageHandler(string packageSource, Common.ILogger logger)
+        public NuGetPackageHandler(Common.ILogger logger, string remotePackageSource, string localPackageDestination = null)
```

Comment:
Don't use optional parameters of public classes and methods (the .Net guidelines say not to because they don't work the same way in all .Net languages). The only place we use them is in test code.
Provide an constructor overload instead.